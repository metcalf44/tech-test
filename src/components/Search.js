
import React, { useState } from "react";
import "../styles/Search.css";
import getImages from "../requests/getImages";
import PropTypes from "prop-types"

const Search = ({ setSearchResults, setSearched }) => {
    const [value, setValue] = useState();

    const handleSubmit = async (event) => {
        event.preventDefault();
        setSearchResults(await getImages(value));
        setSearched(true);
    };

    return (
        <>
            <form className="search-form" onSubmit={handleSubmit}>
                <input 
                className="search-input" 
                type="text" 
                onChange={(e) => setValue(e.target.value)}
                placeholder="Search for"
                data-testid="search-input"
                />
                <button 
                className="search-btn" 
                type="submit"
                data-testid="search-btn"
                >
                    Display
                </button>
            </form>
        </>
    );
};

Search.propTypes = {
    setSearchResults: PropTypes.func.isRequired,
    setSearched: PropTypes.func.isRequired,
};

export default Search