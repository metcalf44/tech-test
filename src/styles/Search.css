
.search-input {
    width: 50%;
    margin-top: 0;
    border-radius: 15px;
    font-size: 1.5rem;
    padding: .3em .5em;
    text-align: center;
    background: rgb(241, 241, 241);
    max-width: 100%;
}

.search-form {
    width: 80%;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    margin: auto; 
    padding-bottom: 3em;
}

.search-btn {
    border-radius: .75em;
    font-weight: 300;
    font-size: 1.25rem;
    text-align: center;
    transition: all .2s;
    width: 35%;
    background: #c4021e;
    color: #fff;
    max-width: 100%;
    padding: .25em .5em;
    box-shadow: 2px 3px rgb(196, 196, 196);
}

.search-btn:hover {
    color: #fff;
    background: #fa2041;
}

.search-btn:focus {
    color: #fff;
    background: #fa2041;
}

.search-input:hover {
    color: #000;
    border: 2px solid #c4021e;
}
.search-input:focus {
    color: #000;
    border: 2px solid #c4021e;
}

@media only screen and (min-width: 760px) and (max-width:1200px) {
    .search-form {
       padding-bottom: 5.5em;
    }
}
@media only screen and (min-width: 1201px) {
    .search-form {
        padding-bottom: 6.5em;
     }
    .search-btn {
        font-size: 1.75rem;
    }

    .search-input {
        font-size: 2em;
    }
}