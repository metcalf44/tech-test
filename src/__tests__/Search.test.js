
import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("<Search />", () => {

    const validProps = {
        setSearchResults: () => {}
    };

    it("renders correctly", () => {
        const { asFragment } = render(
        <Search setSearchResults={validProps.setSearchResults} />
        );
        expect(asFragment()).toMatchSnapshot();
     });
 
 it("renders input correctly", () => {
     const { getByTestId } = render(
     <Search setSearchResults={validProps.setSearchResults} />
     );
     const input = getByTestId("search-input");
     expect(input).toBeTruthy();
 });

 it("renders button correctly", () => {
     const { getByTestId } = render(
         <Search setSearchResults={validProps.setSearchResults} />
     );
     const button = getByTestId("search-btn");
     expect(button).toBeTruthy();
 })
 
 });
