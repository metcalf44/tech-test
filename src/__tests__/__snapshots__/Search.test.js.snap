// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      data-testid="search-input"
      placeholder="Search for"
      type="text"
    />
    <button
      class="search-btn"
      data-testid="search-btn"
      type="submit"
    >
      Display
    </button>
  </form>
</DocumentFragment>
`;
